PROTO_tests/tests/pdarray_creation_test.py

```diff
@@ -0,0 +1,882 @@
+        assert isinstance(pda, ak.pdarray)
+        assert 10 == len(pda)
+        assert int == pda.dtype
+
+        av = ak.array(np.array([[0, 1], [0, 1]]))
+        assert isinstance(av, ak.ArrayView)
+
+        with pytest.raises(TypeError):
+            ak.array({range(0, 100)})
+
+        with pytest.raises(TypeError):
+            ak.array("not an iterable")
+
+        with pytest.raises(TypeError):
+            ak.array(list(list(0)))
+
+    def test_bigint_creation(self):
+        bi = 2**200
+
+        pda_from_str = ak.array([f"{i}" for i in range(bi, bi + 10)], dtype=ak.bigint)
+        pda_from_int = ak.array([i for i in range(bi, bi + 10)])
+        cast_from_segstr = ak.cast(ak.array([f"{i}" for i in range(bi, bi + 10)]), ak.bigint)
+        for pda in [pda_from_str, pda_from_int, cast_from_segstr]:
+            assert isinstance(pda, ak.pdarray)
+            assert 10 == len(pda)
+            assert ak.bigint == pda.dtype
+            assert pda[-1] == bi + 9
+
+        # test array and arange infer dtype
+        assert (
+            ak.array([bi, bi + 1, bi + 2, bi + 3, bi + 4]).to_list() == ak.arange(bi, bi + 5).to_list()
+        )
+
+        # test that max_bits being set results in a mod
+        assert ak.arange(bi, bi + 5, max_bits=200).to_list() == ak.arange(5).to_list()
+
+        # test ak.bigint_from_uint_arrays
+        # top bits are all 1 which should be 2**64
+        top_bits = ak.ones(5, ak.uint64)
+        bot_bits = ak.arange(5, dtype=ak.uint64)
+        two_arrays = ak.bigint_from_uint_arrays([top_bits, bot_bits])
+        assert ak.bigint == two_arrays.dtype
+        assert two_arrays.to_list() == [2**64 + i for i in range(5)]
+        # top bits should represent 2**128
+        mid_bits = ak.zeros(5, ak.uint64)
+        three_arrays = ak.bigint_from_uint_arrays([top_bits, mid_bits, bot_bits])
+        assert three_arrays.to_list() == [2**128 + i for i in range(5)]
+
+        # test round_trip of ak.bigint_to/from_uint_arrays
+        t = ak.arange(bi - 1, bi + 9)
+        t_dup = ak.bigint_from_uint_arrays(t.bigint_to_uint_arrays())
+        assert t.to_list() == t_dup.to_list()
+        assert t_dup.max_bits == -1
+
+        # test setting max_bits after creation still mods
+        t_dup.max_bits = 200
+        assert t_dup.to_list() == [bi - 1, 0, 1, 2, 3, 4, 5, 6, 7, 8]
+
+        # test slice_bits along 64 bit boundaries matches return from bigint_to_uint_arrays
+        for i, uint_bits in enumerate(t.bigint_to_uint_arrays()):
+            slice_bits = t.slice_bits(64 * (4 - (i + 1)), 64 * (4 - i) - 1)
+            assert uint_bits.to_list() == slice_bits.to_list()
+
+    def test_arange(self):
+        assert [0, 1, 2, 3, 4] == ak.arange(0, 5, 1).to_list()
+        assert [5, 4, 3, 2, 1] == ak.arange(5, 0, -1).to_list()
+        assert [-5, -6, -7, -8, -9] == ak.arange(-5, -10, -1).to_list()
+        assert [0, 2, 4, 6, 8] == ak.arange(0, 10, 2).to_list()
```

Comment:
Update these assertions to test against the `NumPy` equivalent functionality instead of hardcoded lists.